spec/controllers/collection_profile_controller_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+describe CollectionProfileController do
+
+  it "checks to see if the collection exists" do
```

Comment:
I am not an expert, but I think it's generally consider good practice to describe the action, context, and expected behavior like: 

```
describe CollectionProfileController do
  describe "GET #show" do
    context "collection does not exist" do
      it "redirects and provides an error message" do
        get :show, collection_id: "A non existent collection"
        expect(response).to redirect_to(collections_path)
        expect(flash[:error]).to eq "What collection did you want to look at?"
      end
    end
  end
end
```

So if someone were to extend this test, they could do:
```
describe CollectionProfileController do
  describe "GET #show" do
    context "collection does not exist" do
      it "redirects and provides an error message"
        get :show, collection_id: "A non existent collection"
        expect(response).to redirect_to(collections_path)
        expect(flash[:error]).to eq "What collection did you want to look at?"
      end
    end

    context "collection exists" do
      let(:collection_profile) { FactoryGirl.create etc }
      it "renders the show template" do
        get :show, collection_id: collection_profile.something
        expect(response).to render_template("show")
      end
    end
  end

  describe "PUT #update" do
    context "attributes are valid" do
      it "changes @collection_profile's attributes" do
        expect something
      end
    end
  end

end
```